docs/typescript.md

```diff
@@ -1,14 +1,637 @@
 # TypeScript Guidelines
 
-These guidelines specifically apply to TypeScript.
+## Introduction
 
-## Provide explicit return types for functions/methods
+These guidelines apply to TypeScript code in MetaMask repos.
+
```

Comment:
This sentence was originally an analogue to the sentence at the beginning of the JavaScript guidelines, but I had been meaning to remove it as it's kind of redundant. Thoughts on removing?

``` suggestion
```